File: Dealing with Missing or Incorrect Data Types.py
import pandas as pd

Eating_Habits = pd.read_excel('C:/Users/olver/Desktop/Eating_Habits.xlsx')

#Recode Activity into a new variable. Zeros should be not eating chocolate, and 1s should be eating chocolate.
def Activity (series): 
    if series == "Eating candy, chocolate bars":
        return 1
    else: 
        return 0
Eating_Habits['ChocolateYN'] = Eating_Habits['Activity'].apply(Activity)

#Recode Frequency & Sex from text to numbers in the same variable. The value zero should be the lowest frequency.
cleanup = {"Frequency" : {"Never" : 0, "Seldom" : 1, "At least once a week" : 2, "Once a day" : 3, "More than once a day" : 4}, "Sex" : {"Females" : 0, "Males" : 1}}
Eating_Habits.replace(cleanup, inplace=True)

#Dummy code the Age group variable.
AgeDummy = pd.get_dummies(Eating_Habits['Age group'], drop_first=True)
AgeDummy

EatingHabits2 = pd.concat([Eating_Habits, AgeDummy], axis=1)

File: Dealing with Missing or Incorrect Data Types.R
#Recode Activity into a new variable called JunkFood. Anything that you would consider junk food, recode as a 1. Everything else should be recoded as a zero.

Eating_Habits$JunkFood < NA
Eating_Habits$JunkFood[Eating_Habits$Activity=='Eating fruit'] <- 0
Eating_Habits$JunkFood[Eating_Habits$Activity=='Eating raw vegetables'] <- 0
Eating_Habits$JunkFood[Eating_Habits$Activity=='Eating candy, chocolate bars'] <- 1
Eating_Habits$JunkFood[Eating_Habits$Activity=='Eating potato chips, crisps'] <- 1
Eating_Habits$JunkFood[Eating_Habits$Activity=='Eating french fries'] <- 1
Eating_Habits$JunkFood[Eating_Habits$Activity=='Eating hamburgers, hot dogs or sausages'] <- 1
Eating_Habits$JunkFood[Eating_Habits$Activity=='Eating peanuts'] <- 0
Eating_Habits$JunkFood[Eating_Habits$Activity=='Eating whole wheat or rye bread'] <- 0
Eating_Habits$JunkFood[Eating_Habits$Activity=='Drinking soft drinks, cola or other drinks with sugar'] <- 1
Eating_Habits$JunkFood[Eating_Habits$Activity=='Drinking coffee'] <- 0
Eating_Habits$JunkFood[Eating_Habits$Activity=='Eating Fruit'] <- 0

#Recode Sex from text to numbers in the same variable.

Eating_Habits$Sex[Eating_Habits$Sex=='Males'] <- 0
Eating_Habits$Sex[Eating_Habits$Sex=='Females'] <- 1